graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -130,6 +499,15 @@ def _get_filter_selectivity(
                 # The estimate may be above 1.0 in case of duplicates in the collection
                 # so we make sure the value is <= 1.0
             )
+    elif filter_info.op_name in INEQUALITY_OPERATORS:
+        # TODO(vlad): Since we assume each filter is independent, we don't consider the correlation
+        #             inequality filters often have. For example, a 'between' filter and an
+        #             equivalent pair of '>=' and '<=' are estimated differently. Consult the
+        #             FilterSelectivityTests/test_inequality_filters_on_uuid function for further
+        #             information.
```

Comment:
Just checking if I get this: if you have a uuid filter `> median - 1` and another `< median + 1`, the correct result is `fractional selectivity 1/2^128`, but since you process and combine selectivities separately, you'd get `fractional selectivity 1/4`.

I'm not arguing we should fix it. It's definitely a good idea to leave it as a TODO for now.

Reply:
Yep, exactly. Fixing this requires revising one of our assumptions _filter selectivities combine independently_.